src/realm.js

```diff
@@ -325,7 +325,11 @@ export class Realm {
     return this.evaluateForEffects(() => env.evaluateAbstractCompletion(ast, strictCode), state);
   }
 
-  evaluateNodeForEffectsInGlobalEnv(node: BabelNode, state?: any) {
+  evaluateAndRevertInGlobalEnv(func: () => void): void {
```

Comment:
To fit in with the other functions, rather name this "evaluateForEffectsInGlobalEnv".

Reply:
Should it still be `evaluateForEffects` if this function doesn't return `Effects`?